tests/formats/test_ascii.py

```diff
@@ -434,6 +434,146 @@ async def test_ascii_check_last_page(
+        tmp_path=tmp_path,
+        manifest_type=manifest_type,
+        request=request,
+        full_load=True
+    )
+    app = create_test_client(context)
+    app.authmodel('example/ascii', ['insert', 'getall', 'search'])
+
+    pushdata(app, f'/example/ascii/text/Country', {
+        'id': 0,
+        'name': {
+            'lt': 'Lietuva',
+            'en': 'Lithuania',
+            'C': 'LT'
+        }
+    })
+    pushdata(app, f'/example/ascii/text/Country', {
+        'id': 1,
+        'name': {
+            'lt': 'Anglija',
+            'en': 'England',
+            'C': 'UK'
+        }
+    })
+
+    res = app.get("/example/ascii/text/Country/:format/ascii?select(id,name)", headers=Headers(headers={
+        'accept-language': 'lt'
+    })).text
+    assert res == (
+        '--  -------\n'
+        'id  name   \n'
+        '0   Lietuva\n'
+        '1   Anglija\n'
+        '--  -------\n'
+    )
+
+
+@pytest.mark.manifests('internal_sql', 'csv')
+def test_ascii_text_with_lang(
+    manifest_type: str,
+    tmp_path: Path,
+    rc: RawConfig,
+    postgresql: str,
+    request: FixtureRequest,
+):
+    context = bootstrap_manifest(
+        rc, '''
+    d | r | b | m | property | type    | ref     | access  | level | uri
+    example/ascii/text/lang    |         |         |         |       | 
+      |   |   |   |          | prefix  | rdf     |         |       | http://www.rdf.com
+      |   |   |   |          |         | pav     |         |       | http://purl.org/pav/
+      |   |   |   |          |         | dcat    |         |       | http://www.dcat.com
+      |   |   |   |          |         | dct     |         |       | http://dct.com
+      |   |   | Country      |         | name    |         |       | 
+      |   |   |   | id       | integer |         |         |       |
+      |   |   |   | name     | text    |         | open    | 3     |
+      |   |   |   | name@en  | string  |         | open    |       |
+      |   |   |   | name@lt  | string  |         | open    |       |
+    ''',
+        backend=postgresql,
+        tmp_path=tmp_path,
+        manifest_type=manifest_type,
+        request=request,
+        full_load=True
+    )
+    app = create_test_client(context)
+    app.authmodel('example/ascii', ['insert', 'getall', 'search'])
+
+    pushdata(app, f'/example/ascii/text/lang/Country', {
+        'id': 0,
+        'name': {
+            'lt': 'Lietuva',
+            'en': 'Lithuania',
+            'C': 'LT'
+        }
+    })
+    pushdata(app, f'/example/ascii/text/lang/Country', {
+        'id': 1,
+        'name': {
+            'lt': 'Anglija',
+            'en': 'England',
+            'C': 'UK'
+        }
+    })
+
+    res = app.get("/example/ascii/text/lang/Country/:format/ascii?lang(*)&select(id,name)", headers=Headers(headers={
+        'accept-language': 'lt'
+    })).text
+    assert res == (
+        '--  ------  ---------  -------\n'
+        'id  name.C  name.en    name.lt\n'
```

Comment:
Would it be difficult to change column names into:

```
id name@C name@en name@lt
```

To match `property` values in manifest table? Not only for ASCII, but for all tabular formats?

Reply:
Currently headers are tightly coupled with mapping logic, in most cases data is given as `dict` format (with no information about types). This `dict` is then flattened and mapped with header names. Changing header names to `name@en` would be easy, but then we get issues with mapping, since flattening function only accepts one type of separator and by default its `.`. In order for it to be able to flatten based on property type, we would need to change how `flatten` works.

Reply:
Moved to a separate issue:

- https://github.com/atviriduomenys/spinta/issues/581